loader/loader.c

```diff
@@ -2937,51 +2953,87 @@ static VkResult read_data_files_in_search_paths(const struct loader_instance *in
-            if (default_config_home == NULL) {
-                vk_result = VK_ERROR_OUT_OF_HOST_MEMORY;
-                goto out;
+    char *home_data_dir = NULL;
+    char *home_config_dir = NULL;
+
+    if (!is_high_integrity()) {
+        // Only use HOME if XDG_DATA_HOME is not present on the system
+        home = loader_secure_getenv("HOME", inst);
+        if (home != NULL) {
+            if (NULL == xdg_config_home || '\0' == xdg_config_home[0]) {
+                const char config_suffix[] = "/.config";
+                default_config_home =
+                    loader_instance_heap_alloc(inst, strlen(home) + strlen(config_suffix) + 1, VK_SYSTEM_ALLOCATION_SCOPE_COMMAND);
+                if (default_config_home == NULL) {
+                    vk_result = VK_ERROR_OUT_OF_HOST_MEMORY;
+                    goto out;
+                }
+                strcpy(default_config_home, home);
+                strcat(default_config_home, config_suffix);
+            }
+            if (NULL == xdg_data_home || '\0' == xdg_data_home[0]) {
+                const char data_suffix[] = "/.local/share";
+                default_data_home =
+                    loader_instance_heap_alloc(inst, strlen(home) + strlen(data_suffix) + 1, VK_SYSTEM_ALLOCATION_SCOPE_COMMAND);
+                if (default_data_home == NULL) {
+                    vk_result = VK_ERROR_OUT_OF_HOST_MEMORY;
+                    goto out;
+                }
+                strcpy(default_data_home, home);
+                strcat(default_data_home, data_suffix);
             }
-            strcpy(default_config_home, home);
-            strcat(default_config_home, config_suffix);
         }
-        if (NULL == xdg_data_home || '\0' == xdg_data_home[0]) {
-            const char data_suffix[] = "/.local/share";
-            default_data_home =
-                loader_instance_heap_alloc(inst, strlen(home) + strlen(data_suffix) + 1, VK_SYSTEM_ALLOCATION_SCOPE_COMMAND);
-            if (default_data_home == NULL) {
-                vk_result = VK_ERROR_OUT_OF_HOST_MEMORY;
-                goto out;
-            }
-            strcpy(default_data_home, home);
-            strcat(default_data_home, data_suffix);
+
+        if (NULL != default_config_home) {
+            home_config_dir = default_config_home;
+        } else {
+            home_config_dir = xdg_config_home;
+        }
+        if (NULL != default_data_home) {
+            home_data_dir = default_data_home;
+        } else {
+            home_data_dir = xdg_data_home;
         }
     }
 #endif  // !_WIN32
 
+    switch (json_type) {
+        case VULKAN_LOADER_JSON_DRIVER:
+            is_icd = true;
+            if (!is_high_integrity()) {
+                override_env = loader_secure_getenv(VK_DRIVER_FILES_ENV_VAR, inst);
+                if (NULL == override_env) {
+                    // Not there, so fall back to the old name
+                    override_env = loader_secure_getenv(VK_ICD_FILENAMES_ENV_VAR, inst);
```

Comment:
Is it well documented that if both env-vars are present, only VK_DRIVER_FILES is used.

Reply:
Very good point.  I'll update the docs.